tests/test_vega.py

```diff
@@ -263,7 +256,776 @@ def test_fill_anchor_in_string(tmp_dir):
+                    "other": "field",
+                    "filename": "test",
+                    "field": "acc",
+                    "loss": 2,
+                },
+            ],
+            "acc",
+            [{"filename": "test", "field": "acc"}],
+            ["rev", "acc", "other", "loss"],
+            {},
+            [{"field": "rev"}, {"field": "loss"}, {"field": "acc"}],
+        ),
+        (
+            [
+                {
+                    "rev": "B",
+                    "dvc_inferred_y_value": "0.05",
+                    "test_acc": "0.05",
+                    "train_acc": "0.06",
+                    "other": "field",
+                    "filename": "data",
+                    "field": "test_acc",
+                    "loss": 0.1,
+                },
+                {
+                    "rev": "B",
+                    "dvc_inferred_y_value": "0.06",
+                    "test_acc": "0.05",
+                    "train_acc": "0.06",
+                    "other": "field",
+                    "filename": "data",
+                    "field": "train_acc",
+                    "loss": 0.1,
+                },
+                {
+                    "rev": "C",
+                    "dvc_inferred_y_value": "0.1",
+                    "train_acc": "0.1",
+                    "test_acc": "0.2",
+                    "other": "field",
+                    "filename": "train_acc",
+                    "field": "acc",
+                    "loss": 2,
+                },
+                {
+                    "rev": "C",
+                    "dvc_inferred_y_value": "0.2",
+                    "train_acc": "0.1",
+                    "test_acc": "0.2",
+                    "other": "field",
+                    "filename": "test_acc",
+                    "field": "acc",
+                    "loss": 2,
+                },
+            ],
+            "dvc_inferred_y_value",
+            [
+                {"filename": "data", "field": "train_acc"},
+                {"filename": "data", "field": "test_acc"},
+            ],
+            ["rev", "dvc_inferred_y_value", "train_acc", "test_acc", "other", "loss"],
+            {
+                "field": "field",
+                "legend": {
+                    "symbolFillColor": "transparent",
+                    "symbolStrokeColor": "grey",
+                },
+                "scale": {
+                    "domain": ["test_acc", "train_acc"],
+                    "range": ["circle", "square"],
+                },
+            },
+            [
+                {"field": "rev"},
+                {"field": "loss"},
+                {"field": "dvc_inferred_y_value"},
+                {"field": "field"},
+            ],
+        ),
+        (
+            [
+                {
+                    "rev": "B",
+                    "dvc_inferred_y_value": "0.05",
+                    "test_acc": "0.05",
+                    "other": "field",
+                    "filename": "test",
```

Comment:
Can we also have separate test cases where train/test vary by field OR by filename (not both)?